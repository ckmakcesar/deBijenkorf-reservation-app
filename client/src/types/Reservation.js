import * as PropTypes from 'prop-types';

const Reservation = {
  id: PropTypes.number.isRequired,
  storeId: PropTypes.number.isRequired,
  statusId: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  createdAt: PropTypes.string.isRequired,
};

export default Reservation;
