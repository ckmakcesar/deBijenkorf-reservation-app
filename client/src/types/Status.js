import * as PropTypes from 'prop-types';

const Status = {
  id: PropTypes.number.isRequired,
  code: PropTypes.string.isRequired,
};

export default Status;
