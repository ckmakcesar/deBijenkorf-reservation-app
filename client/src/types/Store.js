import * as PropTypes from 'prop-types';

const Store = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  address: PropTypes.string.isRequired,
};

export default Store;
