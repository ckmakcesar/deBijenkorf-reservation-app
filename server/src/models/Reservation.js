const { Sequelize, DataTypes } = require('sequelize');

const { sequelize } = require('.');

const Reservation = sequelize.define('reservation', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  name: { type: DataTypes.STRING, allowNull: false },
  storeId: { type: DataTypes.INTEGER, allowNull: false },
  statusId: { type: DataTypes.INTEGER, allowNull: false },
  time: { type: DataTypes.DATEONLY, allowNull: false },
  createdAt: { type: DataTypes.DATE, allowNull: false, defaultValue: Sequelize.NOW },
}, {
  freezeTableName: true,
  underscored: true,
  version: false,
  updatedAt: false,
});

module.exports = Reservation;
