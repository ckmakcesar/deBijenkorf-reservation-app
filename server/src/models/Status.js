const { DataTypes } = require('sequelize');

const { sequelize } = require('.');

const Status = sequelize.define('status', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  code: { type: DataTypes.STRING, allowNull: false, unique: true },
  name: { type: DataTypes.STRING, allowNull: false, unique: true },
}, {
  freezeTableName: true,
  underscored: true,
  version: false,
  createdAt: false,
  updatedAt: false,
});

module.exports = Status;
