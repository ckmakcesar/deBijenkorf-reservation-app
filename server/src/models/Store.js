const { DataTypes } = require('sequelize');

const { sequelize } = require('.');
const Reservation = require('./Reservation');

const Store = sequelize.define('status', {
  id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
  name: { type: DataTypes.STRING, allowNull: false, unique: true },
  address: { type: DataTypes.STRING, allowNull: false },
}, {
  freezeTableName: true,
  underscored: true,
  version: false,
  createdAt: false,
  updatedAt: false,
});

// Associations
Store.hasMany(Reservation, {
  sourceKey: 'id',
  foreignKey: 'storeId',
  as: 'reservations',
})
Reservation.belongTo(Store, {
  foreignKey: 'storeId',
  as: 'store',
})

module.exports = Store;
