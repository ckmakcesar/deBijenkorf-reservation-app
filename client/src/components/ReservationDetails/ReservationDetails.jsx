import React, { useState } from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';

import Button from '../misc/Button';
import Reservation from '../../types/Reservation';
import Store from '../../types/Store';
import Status from '../../types/Status';
import { formatDate } from '../../utils/utils';

import { reservationStatusCodesToColors } from '../../constants';
import styles from '../../styles/ReservationDetails.module.css';

const ReservationDetails = ({
  reservation,
  store,
  status,
  setReservationToDelete,
  setDrawerReservationId,
}) => {
  const [hover, setHover] = useState(false);

  const handleClick = () => {
    setDrawerReservationId(reservation.id);
  };
  const handleMouseEnter = () => {
    setHover(true);
  };
  const handleMouseLeave = () => {
    setHover(false);
  };

  return (
    <div
      className={clsx(styles.reservationDetailsContainer, hover && styles.reservationDetailsHover)}
      onClick={handleClick}
      onMouseEnter={handleMouseEnter}
      onMouseLeave={handleMouseLeave}
    >
      <div className={styles.reservationDetailsLeft}>
        <div className={styles.reservationDetailsName}>
          {reservation.name}
        </div>
        <div className={styles.reservationDetailsContext}>
          {formatDate(reservation.date)}
          &nbsp;Â·&nbsp;
          {store.name}
        </div>
      </div>

      <div className={styles.reservationDetailsRight}>
        <div className={clsx(styles.reservationDetailsStatus, reservationStatusCodesToColors[status.code])}>
          {status.name}
        </div>
        <Button
          onClick={(e) => {
            e.stopPropagation(); // not to trigger "click row for viewing/editing details"
            setReservationToDelete(reservation);
          }}
          text='Delete'
          boxIconClassName='bx-trash'
          tone='red'
        />
      </div>
    </div>
  );
};

// *********************************************************************************************************************
// PROPS
// *********************************************************************************************************************
ReservationDetails.propTypes = {
  reservation: PropTypes.exact(Reservation).isRequired,
  store: PropTypes.exact(Store),
  status: PropTypes.exact(Status),
  setReservationToDelete: PropTypes.func.isRequired,
  setDrawerReservationId: PropTypes.func.isRequired,
};

ReservationDetails.defaultProps = {
  store: {},
  status: {},
};

export default ReservationDetails;
