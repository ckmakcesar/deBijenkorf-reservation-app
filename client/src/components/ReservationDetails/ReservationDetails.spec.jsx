import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ReservationDetails from './ReservationDetails';
import { mockReservations } from '../../tests/data/Reservation';
import { mockStores } from '../../tests/data/Store';
import { mockStatuses } from '../../tests/data/Status';

describe('component:ReservationDetails', () => {
  it('Should render ReservationDetails with every optional props present', () => {
    const wrapper = shallow(
      <ReservationDetails
        reservation={mockReservations[0]}
        store={mockStores[0]}
        status={mockStatuses[0]}
        setReservationToDelete={jest.fn()}
        setDrawerReservationId={jest.fn()}
      />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
