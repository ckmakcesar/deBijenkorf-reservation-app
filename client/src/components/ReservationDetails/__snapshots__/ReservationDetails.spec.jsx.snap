// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component:ReservationDetails Should render ReservationDetails with every optional props present 1`] = `
<div
  className="reservationDetailsContainer"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="reservationDetailsLeft"
  >
    <div
      className="reservationDetailsName"
    >
      Repair
    </div>
    <div
      className="reservationDetailsContext"
    >
      2022-01-27
       · 
      Centraal
    </div>
  </div>
  <div
    className="reservationDetailsRight"
  >
    <div
      className="reservationDetailsStatus green"
    >
      Ready
    </div>
    <Button
      boxIconClassName="bx-trash"
      disabled={false}
      formId=""
      isSubmit={false}
      onClick={[Function]}
      text="Delete"
      tone="red"
    />
  </div>
</div>
`;
